contracts/multisig/test/asset_proxy_owner.ts

```diff
@@ -1,511 +1,641 @@
-            expect(isErc20ProxyRegistered).to.equal(true);
-            expect(isErc721ProxyRegistered).to.equal(true);
-        });
-        it('should revert if a null address is included in assetProxyContracts', async () => {
-            const assetProxyContractAddresses = [erc20Proxy.address, constants.NULL_ADDRESS];
-            return expectContractCreationFailedAsync(
-                (AssetProxyOwnerContract.deployFrom0xArtifactAsync(
-                    artifacts.AssetProxyOwner,
-                    provider,
-                    txDefaults,
-                    artifacts,
-                    owners,
-                    assetProxyContractAddresses,
-                    REQUIRED_APPROVALS,
-                    SECONDS_TIME_LOCKED,
-                ) as any) as sendTransactionResult,
-                RevertReason.InvalidAssetProxy,
+                reg.functionSelectors,
+                reg.destinations,
+                reg.functionCallTimeLockSeconds,
+                signerAddresses,
+                new BigNumber(REQUIRED_SIGNERS),
+                new BigNumber(DEFAULT_TIME_LOCK),
             );
+            await expect(tx).to.revertWith(RevertReason.EqualLengthsRequired);
+        });
+        it('should allow no function calls to be registered', async () => {
+            const tx = TestAssetProxyOwnerContract.deployFrom0xArtifactAsync(
+                artifacts.TestAssetProxyOwner,
+                env.provider,
+                env.txDefaults,
+                artifacts,
+                [],
+                [],
+                [],
+                signerAddresses,
+                new BigNumber(REQUIRED_SIGNERS),
+                new BigNumber(DEFAULT_TIME_LOCK),
+            );
+            expect(tx).to.be.fulfilled('');
+        });
+        it('should register a single functon call', async () => {
+            const reg = createFunctionRegistration(1, 1, 1);
+            const apOwner = await TestAssetProxyOwnerContract.deployFrom0xArtifactAsync(
+                artifacts.TestAssetProxyOwner,
+                env.provider,
+                env.txDefaults,
+                artifacts,
+                reg.functionSelectors,
+                reg.destinations,
+                reg.functionCallTimeLockSeconds,
+                signerAddresses,
+                new BigNumber(REQUIRED_SIGNERS),
+                new BigNumber(DEFAULT_TIME_LOCK),
+            );
+            const timelock = await apOwner.functionCallTimeLocks.callAsync(
+                reg.functionSelectors[0],
+                reg.destinations[0],
+            );
+            expect(timelock[0]).to.equal(true);
+            expect(timelock[1]).to.bignumber.equal(reg.functionCallTimeLockSeconds[0]);
+        });
+        it('should register multiple function calls', async () => {
+            const reg = createFunctionRegistration(2, 2, 2);
+            const apOwner = await TestAssetProxyOwnerContract.deployFrom0xArtifactAsync(
```

Comment:
Optional: It would be pretty simple to take the logic from this function and the above function to create a parameterized unit test for different numbers of registered functions. I'm not sure how important fuzzing is in this case, but it would be nice to have the option open and to add unit tests up to 10 registered functions.